rollup/core/classes/RollupParentResetProcessor.cls

```diff
@@ -71,31 +87,34 @@ public without sharing class RollupParentResetProcessor extends RollupFullBatchR
     }
     this.logger.log('resetting parent fields for: ' + parentItems.size() + ' items', System.LoggingLevel.INFO);
     Map<String, Schema.SObjectField> parentFields = parentItems.get(0).getSObjectType().getDescribe().fields.getMap();
-    for (SObject parentItem : parentItems) {
-      for (Rollup__mdt rollupMeta : this.rollupMetas) {
-        if (
-          rollupMeta.LookupObject__c == this.calcItemType.getDescribe().getName() &&
-          this.parentRollupFieldHasBeenReset(
-            rollupMeta.LookupFieldOnLookupObject__c,
-            parentItem,
-            rollupMeta.LookupObject__c,
-            rollupMeta.RollupFieldOnLookupObject__c
-          ) == false &&
-          parentFields.containsKey(rollupMeta.RollupFieldOnLookupObject__c)
-        ) {
-          parentItem.put(rollupMeta.RollupFieldOnLookupObject__c, this.getDefaultValue(rollupMeta));
-        }
+    for (Rollup__mdt rollupMeta : this.rollupMetas) {
+      Object defaultVal = this.getDefaultValue(rollupMeta);
+      Schema.SObjectField parentFieldToken = parentFields.get(rollupMeta.RollupFieldOnLookupObject__c);
+      if (
+        rollupMeta.LookupObject__c != this.calcItemType.getDescribe().getName() ||
+        this.parentRollupFieldHasBeenReset(rollupMeta.LookupObject__c, rollupMeta.RollupFieldOnLookupObject__c) ||
+        parentFieldToken == null
+      ) {
+        continue;
+      }
+      for (SObject parentItem : parentItems) {
+        this.getDML().updateField(parentFieldToken, parentItem, defaultVal);
```

Comment:
I think you could move the declaration of `defaultVal` to be after your `continue` statement - right now, it's always being retrieved, but it's only used if you make it past the `continue` statement.

Reply:
Perfect, good call. I can't believe it's taken me this long to promote that variable out of the for loop as it is!